pkg/informer/informer.go

```diff
@@ -219,15 +249,115 @@ func (d *scopedDynamicDiscoverySharedInformerFactory) ForResource(gvr schema.Gro
 	return clusterInformer.Cluster(d.cluster), nil
 }
 
+// NewSingleClusterDynamicDiscoverySharedInformerFactory returns a factory for
+// cluster-unaware (standard Kube) shared informers that discovers new types
+// and informs on updates to resources of those types.
+// It retrieves GVR-related information from CRD informers.
+func NewSingleClusterDynamicDiscoverySharedInformerFactory(
+	cfg *rest.Config,
+	filterFunc func(obj interface{}) bool,
+	crdInformer kcpapiextensionsv1informers.CustomResourceDefinitionClusterInformer,
```

Comment:
if this is cluster-unaware, why are we taking a cluster-aware informer?

Reply:
fixed